app/models/spree/admin/resources/product_default_tabs_builder.rb

```diff
@@ -0,0 +1,214 @@
+            url: ->(resource) { admin_product_images_path(resource) },
+            classes: 'nav-link',
+            partial_name: :images
+          }
+        end
+
+        def add_variants_tab(root)
+          tab =
+            Tab.new(variants_config).
+            with_active_check.
+            with_default_translator.
+            with_availability_check(
+              lambda do |ability, resource|
+                ability.can?(:admin, ::Spree::Variant) && !resource.deleted?
+              end
+            )
+
+          root.add(tab)
+        end
+
+        def variants_config
+          {
+            icon_name: 'adjust.svg',
+            name: :variants,
+            url: ->(resource) { admin_product_variants_path(resource) },
+            classes: 'nav-link',
+            partial_name: :variants
+          }
+        end
+
+        def add_properties_tab(root)
+          tab =
+            Tab.new(properties_config).
+            with_active_check.
+            with_default_translator.
+            with_availability_check(
+              lambda do |ability, resource|
+                ability.can?(:admin, ::Spree::ProductProperty) && !resource.deleted?
+              end
+            )
+
+          root.add(tab)
+        end
+
+        def properties_config
+          {
+            icon_name: 'list.svg',
+            name: :properties,
+            url: ->(resource) { admin_product_product_properties_path(resource) },
+            classes: 'nav-link',
+            partial_name: :properties
+          }
+        end
+
+        def add_stock_tab(root)
+          tab =
+            Tab.new(stock_config).
+            with_active_check.
+            with_default_translator.
+            with_availability_check(
+              lambda do |ability, resource|
+                ability.can?(:admin, ::Spree::StockItem) && !resource.deleted?
+              end
+            )
+
+          root.add(tab)
+        end
+
+        def stock_config
+          {
+            icon_name: 'box-seam.svg',
+            name: :stock,
+            url: ->(resource) { stock_admin_product_path(resource) },
+            classes: 'nav-link',
+            partial_name: :stock
+          }
+        end
+
+        def add_prices_tab(root)
+          tab =
+            Tab.new(prices_config).
+            with_active_check.
+            with_default_translator.
+            with_availability_check(
+              lambda do |ability, resource|
+                ability.can?(:admin, ::Spree::Price) && !resource.deleted?
+              end
+            )
+
+          root.add(tab)
+        end
+
+        def prices_config
+          {
+            icon_name: 'currency-exchange.svg',
+            name: :prices,
+            url: ->(resource) { admin_product_prices_path(resource) },
+            classes: 'nav-link',
+            partial_name: :prices
+          }
+        end
+
+        def add_digitals_tab(root)
+          tab =
+            Tab.new(digitals_config).
```

Comment:
In general, maybe we could have two classes - `Tab` (represents the tab) and `TabBuilder` (ensures proper construction of a tab and provides methods to configure it correcly)? Currently `Tab` has a lot of logic related to the construction of a valid `Tab` object. That'd be consistent with the decision we've made for MainMenu sections, and would ensure fairly consistent interfaces across the admin panel configuration features.

Reply:
I initially thought that it'd blur the picture but the separation of concerns buys me. Thank you for that remark. 🙂